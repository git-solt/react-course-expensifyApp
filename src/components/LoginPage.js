import React from 'react'
import {connect} from 'react-redux'
import {startLogin } from '../actions/auth'

const LoginPage = ({startLogin}) => (
  <div>
    <button onClick={startLogin}>Sign in</button>
    <p>Don't have an account? Sing up <a href="">here</a></p>
  </div>
)

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin()) 
})

export default connect(undefined, mapDispatchToProps)(LoginPage)