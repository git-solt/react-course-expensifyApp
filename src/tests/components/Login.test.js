import React from 'react'
import LoginPage from '../../components/LoginPage'
import { shallow } from 'enzyme'

test('should render login page', () => {
  const component = shallow(<LoginPage />)

  expect(component).toMatchSnapshot()
})

test('should call startlogin', () => {
  const startLogin = jest.fn()
  const wrapper = shallow(<LoginPage startLogin={startLogin}/>)
  wrapper.find('button').simulate('click')
  expect(startLogin).toHaveBeenCalled()
})