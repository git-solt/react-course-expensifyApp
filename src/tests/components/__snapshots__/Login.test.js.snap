// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render login page 1`] = `
<div>
  <button>
    Sign in
  </button>
  <p>
    Don't have an account? Sing up 
    <a
      href=""
    >
      here
    </a>
  </p>
</div>
`;
